src/main/kotlin/graphics/scenery/backends/ShaderCompiler.kt

```diff
@@ -187,16 +203,42 @@ class ShaderCompiler: AutoCloseable, Callable<ByteArray> {
             else -> throw UnsupportedOperationException("Unknown shader target $target.")
         }
 
-        println("Compiling $file to $out, type $type, optimising for $level, target $t${if(debug) {", with debug information"} else { "" }}")
-        val bytecode = compile(file.readText(), type, t, entryPoint, debug, strict, level, file.absolutePath, null)
+        val inputName = if(this.file.isDirectory) {
+            file.name
+        } else {
+            file.toString()
+        }
 
-        if(!out.exists()) {
-            out.createNewFile()
+        val outputName = if(this.file.isDirectory) {
+            out.name
         } else {
-            out.writeBytes(bytecode)
+            out.toString()
         }
 
-        return bytecode
+        val code = file.readText()
+        if(code.contains(Regex("#ifdef|#ifndef|#else|#endif"))) {
+            println("⏭️  $inputName -> $outputName SKIPPED ${file.name} due to preprocessor directives in code.")
```

Comment:
what are/were we doing with files with preprocessor directives?

Reply:
That particular function is only used when invoking the compiler from the command line. This means that files with preprocessor directives are just skipped for offline compilation. When invoked from within the renderer, the files are compiled regularly. This is e.g. due to the renderer or volume renderer adding some `#define`s.